### ISSUE TITLE
optimize_radius not using information of clusters

### ISSUE THREAD

**Issue by marco-2023** [theochem/DiverseSelector#154]
The function `optimize_radius` in the file **utils.py** is not using the `cluster_ids`parameters. The `select_from_cluster` method of the `DirectedSphereExclusion` is also not using the  `cluster_ids`. Because of this, this selection algorithm fails when used with cluster labels.

**Comment by xychem** [comment]
The cluster_ids don't be used in ***utils.py***, it seems work when you insert following code to the raw code.

```python
    if cluster_ids is not None:
        # the feature matrix(n_samples,n_features) of different cluster
        X = X[cluster_ids]
```

```python
if X.shape[0] < size:
        raise RuntimeError(
            f"Size of samples to be selected is greater than existing the number of samples; "
            f"{size} > {X.shape[0]}."
        )
    # set the limits on # of selected points according to the tolerance percentage

    if cluster_ids is not None:
        # the feature matrix(n_samples,n_features) of different cluster
        X = X[cluster_ids]

    error = size * obj.tol
    lower_size = round(size - error)
    upper_size = round(size + error)
```

An other question is the ***tol*** of class ***optisim*** in dissimilarity.py is wrong, I think it can be 0.05. The iteration in ***ultis.py*** will not execute due to the huge tolerance.
```python
def __init__(self, r=None, k=10, tol=5.0, eps=0, p=2, start_id=0, random_seed=42, n_iter=10):
```
I get the following pictures after doing these.

## optisim
![image](https://github.com/theochem/DiverseSelector/assets/116488431/9ef885c5-01df-4b11-9e6f-bd725471c16c)
![image](https://github.com/theochem/DiverseSelector/assets/116488431/a807f979-5cd6-4a53-8531-da13b9688076)

## DirectedSphereExclusion 
![image](https://github.com/theochem/DiverseSelector/assets/116488431/5d46f2b8-78e4-4a8d-984e-0de244b37ef5)
#### ***It is strange that I get thirteen points in the second picture but it looks like the points are selected by thier cluster_labels.***
![image](https://github.com/theochem/DiverseSelector/assets/116488431/a2dbf13d-caaf-4f78-88bb-af0f1c5cc974)

**Comment by FarnazH** [comment]
Thanks @xychem, this 183e253db428d0f3abda29231dc01920ace8858a fixes one of the issues you reported in https://github.com/theochem/Selector/issues/154#issue-1843624285. I also encountered it when putting together the `quick_start.ipynb` notebook. Don't know why this item was not addressed before closing the issue.

**Comment by FanwangM** [comment]
This issue was closed because #155 attempted to fix the problem. Accodring to @xychem's code, it seemed working. I will take a closer look at the `quick_start.ipynb`.
